app/api/attendees.py

```diff
@@ -154,6 +156,12 @@ def before_update_object(self, obj, data, kwargs):
                 if obj.checkin_times and data['checkin_times'] not in obj.checkin_times.split(","):
                     data['checkin_times'] = '{},{}'.format(obj.checkin_times, data['checkin_times'])
 
+        if 'is_checked_out' in data and data['is_checked_out']:
```

Comment:
Logic is wrong. Try to simulate things in order to check for logic. 
This will add the checkout_time for every time an update request is made and `is_checked_out` is true. Is this correct ? Absolutely not. You have to check that the value of `is_checked_out` was initially false and then this PATCH request is making it true.